README.md

```diff
@@ -17,44 +18,98 @@ In contrast, our philosophy is rooted in simplicity:
+- Rust.
+- Docker.
+```shell
+make localnet
+```
+
+This make target will:
+1. Build our node inside a docker image.
+2. Fetch our fork [ethereum package](https://github.com/ethpandaops/ethereum-package), a private testnet on which multiple ethereum clients can interact.
+3. Start the localnet with kurtosis.
+
+If everything went well, you should be faced with our client's logs (ctrl-c to leave)
+
+To stop everything, simply run:
+```shell
+make stop-localnet
+```
 
+
+## Dev Setup
 ### Build
 
-To build the main executable and its crates, run:
+#### Rust
+To build the node, you will need the rust toolchain: 
+1. First, [install asdf](https://asdf-vm.com/guide/getting-started.html): 
+2. Add the rust plugin: 
+```shell
+asdf plugin-add rust https://github.com/asdf-community/asdf-rust.git
+```
+3. cd into the project and run:
+```shell
+asdf install
+```
+
+You now should be able to build the client:
 ```bash
 make build
 ```
 
-### Test
-Note: To execute EF tests, the test fixtures are required. To download them, run:
+### Testing
+
+For testing, we're using three kinds of tests.
+
+##### Ethereum Foundation Tests
+
+These are the official execution spec tests, tests that should be implementation agnostic, you can execute them with:
+
 ```bash
-make download-test-vectors
+make test
 ```
 
-To run the tests from a crate, run:
+This will previously download it from the [official execution spec tests repo](https://github.com/ethereum/execution-spec-tests/) and run them with our glue code
+under `cmd/ef_tests/tests/cancun.rs`.
+
+##### Crate Specific Tests
+
+The second kind are each crate tests, you can run it like this:
+
 ```bash
 make test CRATE=<crate>
 ```
 
-Or just run all the tests:
+##### Hive Tests
+
+Finally, we have End-to-End tests with hive.
+Hive is a system which simply sends RPC commands to our node,
+and expects a certain response. You can read more about it [here](https://github.com/ethereum/hive/blob/master/docs/overview.md).
+And you can run this set of tests with:
+
 ```bash
-make test
+make run-hive-debug ethereum/rpc-compat
 ```
 
-### Run
+### Localnet
 
-To run a localnet, we can use a fork of [Ethereum Package](https://github.com/ethpandaops/ethereum-package), specifically [this branch](https://github.com/lambdaclass/ethereum-package/tree/ethereum-rust-integration) that adds support to our client. We have that included in our repo as a `just` target. Make sure to fetch it like follows:
+To run a localnet, we'll use our fork of [ethPandaOps' Ethereum Package](https://github.com/ethpandaops/ethereum-package), specifically [this branch](https://github.com/lambdaclass/ethereum-package/tree/ethereum-rust-integration) that adds support to our client. We have that included in our repo as a `make` target. Make sure to fetch it like follows:
 
 ```bash
 make checkout-ethereum-package
 ```
 
-Let's now install kurtosis:
+If you haven't yet, install Kurtosis:
 
 ```bash
 # Make sure to have docker installed
-
+brew install docker
```

Comment:
Maybe we should link the official docker installation page instead so that users with different systems can also use it? https://docs.docker.com/engine/install/